llvm/lib/Target/AIE/AIELegalizerInfo.cpp

```diff
@@ -1191,6 +1185,123 @@ bool AIELegalizerInfo::legalizeG_FCMP(LegalizerHelper &Helper, MachineInstr &MI,
   return true;
 }
 
+/// @brief Get the AIE2 intrinsic corresponding to the fcmp predicate.
+unsigned getFCmpIntrID(CmpInst::Predicate Predicate, bool &SwapOperands) {
+  switch (Predicate) {
+  default:
+    llvm_unreachable("Unsupported FCmp predicate");
+  case CmpInst::FCMP_OGE:
+  case CmpInst::FCMP_UGE:
+    return Intrinsic::aie2_vgebf16;
```

Comment:
`FCMP_OGE`  and `FCMP_UGE` are not the same. From [LLVM reference manual](https://llvm.org/docs/LangRef.html#id312):
```
oge: yields true if both operands are not a QNAN and op1 is greater than or equal to op2.
uge: yields true if either operand is a QNAN or op1 is greater than or equal to op2.
```

AIE2 floating point comparison instructions have the following behavior:
```
If either of the comparison inputs is NaN, the output is set to zero (false).
```

Thus, we need to do extra work for unordered comparisons, i.e. checking whether one of the inputs is a NaN.
Probably easiest thing to do is to promote the inputs to such comparisons to fp32, then let the existing fp32 custom legalization turn it into libcalls.

Reply:
I changed it to use libcalls for now, since we don't really get unordered comparisons..